### PR TITLE
Module 8: Typescript

### PR DESCRIPTION
**Pull request by louisrli**

### CODE REVIEW

File: module8-typescript/r1.1-classes-and-interfaces/README.md [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+class.
+
+```typescript
+class RecodedMember {
+  getTitle() {
+    return "Generic member";
+  }
+}
+
+class RecodedStudent {
```
Comment (halitbatur) [inline module8-typescript/r1.1-classes-and-interfaces/README.md]: missing the `extends` keyword

File: module8-typescript/r1.1-classes-and-interfaces/README.md [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+ 
+interface ColorfulCircle extends Colorful, Circle {}
+ 
+const cc: ColorfulCircle = {
+  color: "red",
+  radius: 42,
+};
+```
+
+In the example above, we have two iterfaces: `Colorful` and `Circle`. If we want
```
Comment (halitbatur) [inline module8-typescript/r1.1-classes-and-interfaces/README.md]: interfaces*

File: module8-typescript/r2-advanced-types/README.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+}
+```
+
+And they can often be combined with switch-cases:
+```typescript
+function getPrimaryColor(userTheme: Theme) {
+  switch (userTheme) {
+    case Theme.DARK:
+      return "black";
+      return "white";
```
Comment (halitbatur) [inline module8-typescript/r2-advanced-types/README.md]: This line should be after line 46

File: module8-typescript/r2-advanced-types/README.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    default:
+      return "blue";
+  }
+}
+```
+
+## Intersection types
+Previously, we discussed union types, such as `string | undefined`, meaning that
+a value could be either of type `string` or type `undefined`. However, an
+intersection type instead *combines* all of the given types.
```
Comment (halitbatur) [inline module8-typescript/r2-advanced-types/README.md]: no examples were mentioned for intersection types.

**Review by halitbatur — CHANGES_REQUESTED** [review]

**Review by wisammechano — APPROVED** [review]
Made the necessary changes for version 1 of the module

**Comment by wisammechano** [comment]
Moved to Canvas

**Pull request merged**